### PR TITLE
feat: breadcrumbs component

### PR DESCRIPTION
**Pull request by ramonmulia**
PR to implement Breadcrumbs component

added:

- stories
- tests
- macro
- docs
- example in nextjs and wagtail

Screenshot 2024-11-25 at 13 41 41

### CODE REVIEW

File: apps/docs/content/3-components/2-library/4-breadcrumbs.mdx [hunk 1]
```diff
@@ -9,6 +9,61 @@ status: in-development
 
 <ComponentStatusBlock componentId="breadcrumbs" />
 
+<Breadcrumbs>
+  <BreadcrumbLink href="/home">Home</BreadcrumbLink>
+  <BreadcrumbEllipses />
```
Comment (BogdanDanilescu) [inline apps/docs/content/3-components/2-library/4-breadcrumbs.mdx:14]: I think there's a typo. I think it should be `BreadCrumbEllipsis`

**Review by pregno — APPROVED** [review]

**Comment by BogdanDanilescu** [comment]
can you add an example in the wagtail application ?

File: packages/react/ds/src/breadcrumbs/breadcrumbs.tsx [hunk 2]
```diff
@@ -0,0 +1,43 @@
+import { Link } from '../link/link.js';
+import type { BreadcrumbLinkProps, BreadcrumbProps } from './types.js';
+
+export const BreadcrumbEllipsis = () => (
+  <div className="gi-breadcrumb-ellipsis" aria-hidden="true">
```
Comment (hamza14khan) [inline packages/react/ds/src/breadcrumbs/breadcrumbs.tsx:5]: suggestion: Not sure if you have consult this with Lexa or not, but we have an icon for this.

```
<span data-testid="govie-icon" class="material-symbols-outlined gi-gray-700">more_horiz</span>
```

Comment (ramonmulia) [reply packages/react/ds/src/breadcrumbs/breadcrumbs.tsx:5]: hmm I didn't know that, let me check it.

File: packages/html/ds/src/breadcrumbs/breadcrumbs.html [hunk 3]
```diff
@@ -0,0 +1,30 @@
+<nav aria-label="Breadcrumbs" class="gi-breadcrumbs">
```
Comment (pregno) [inline packages/html/ds/src/breadcrumbs/breadcrumbs.html:1]: Could you add a macro to create this structure?
I think we can accept as a param a list of items like this:
```
{
   "label": string;
   "href": string;
   "ellipsis": boolean;
}
```
So this example will be achieved using:
```
{{
   govieBreadcrumbs([
      {
         "label": "Home",
         "href": "/"
      },
      {
         "ellipsis": true
      },
      {
         "label": "Travel",
         "href": "/travel"
      },
      {
         "label": "Documentation",
         "href": "/travel/docs"
      }
   ])
}}
```

**Review by pregno — CHANGES_REQUESTED** [review]

File: packages/react/ds/src/breadcrumbs/breadcrumbs.tsx (hunk 2 above)
Comment (hamza14khan) [reply packages/react/ds/src/breadcrumbs/breadcrumbs.tsx:5]: I have used in my html pagination PR if you want a reference.

**Review by hamza14khan — APPROVED** [review]

**Review by BogdanDanilescu — APPROVED** [review]

**Review by pregno — APPROVED** [review]

**Pull request merged**